src/cliapp.rs

```diff
@@ -26,10 +26,94 @@ pub fn make_app() -> App<'static, 'static> {
                 .global(true)
                 .help("The path to the config folder."),
         )
-        .subcommand(App::new("run").about("Run the relay").after_help(
-            "This runs the relay in the foreground until it's shut down.  It will bind \
+        .subcommand(
+            App::new("run")
+                .about("Run the relay")
+                .after_help(
+                    "This runs the relay in the foreground until it's shut down.  It will bind \
              to the port and network interface configured in the config file.",
-        ))
+                )
+                .arg(
+                    Arg::with_name("secret_key")
+                        .long("secret-key")
+                        .short("s")
+                        .value_name("KEY")
+                        .requires("public_key")
+                        .help("The secret key to set"),
+                )
+                .arg(
+                    Arg::with_name("public_key")
+                        .long("public-key")
+                        .short("p")
+                        .value_name("KEY")
+                        .requires("secret_key")
+                        .help("The public key to set"),
+                )
+                .arg(
+                    Arg::with_name("id")
+                        .long("id")
+                        .short("i")
+                        .value_name("RELAY_ID")
+                        .help("The relay ID to set"),
+                )
+                .arg(
+                    Arg::with_name("upstream-url")
+                        .value_name("UPSTREAM_URL")
+                        .takes_value(true)
+                        .short("u")
+                        .long("upstream-url")
+                        .help("The upstream server URL."),
+                )
+                .arg(
+                    Arg::with_name("host")
+                        .value_name("HOST")
+                        .takes_value(true)
+                        .short("H")
+                        .long("host")
+                        .help("The host dns name."),
+                )
+                .arg(
+                    Arg::with_name("port")
+                        .value_name("PORT")
+                        .takes_value(true)
+                        .short("P")
+                        .long("port")
+                        .help("The server port."),
+                )
+                .arg(
+                    Arg::with_name("processing-enabled")
```

Comment:
Can we shorten this to `--processing`?

Reply:
Not that this is not an optional parameter but a flag, there are two:  --processing-enabled and --processing-disabled it makes it clear what you are specifying, if you want to save your keyboard from wear you can also use -e or -d. 

I think it is nicer to have flags for boolean values rather than parameters with value 
`--processing true`, or `--processing 1` .

Any way if you still think it is a good idea I can change it to either a parameter 
`--processing true | 1`  or a flag  `--processing` and `--no-processing`. 

Note that using --processing will make the short option unintuitive because p is already used twice `-p` for `--public-key` and `-P` for `--port`